pkgs/development/libraries/qt-6/modules/qtbase.nix

```diff
@@ -178,8 +194,13 @@ stdenv.mkDerivation rec {
   inherit patches;
 
   # https://bugreports.qt.io/browse/QTBUG-97568
+  # https://gitlab.kitware.com/cmake/cmake/-/issues/23251
   postPatch = ''
     substituteInPlace src/corelib/CMakeLists.txt --replace /bin/ls ${coreutils}/bin/ls
+    substituteInPlace CMakeLists.txt \
+      --replace "qt_internal_check_if_path_has_symlinks(" "#"
+    substituteInPlace cmake/QtAutoDetect.cmake \
+      --replace "/usr/bin/xcrun --sdk" "echo ${stdenv.targetPlatform.darwinSdkVersion} #"
```

Comment:
FYI, we have a stubbed xcrun under `xcbuild` which prints our SDK version; we could just include that in nativeBuildInputs.

I'm also fine with a substitution like this but it might be less stable.

Reply:
Seems like adding `xcbuild` doesn't work:

```
CMake Error at cmake/QtAutoDetect.cmake:283 (message):
  Can't determine darwin macosx SDK version.  Error:
Call Stack (most recent call first):
  cmake/QtAutoDetect.cmake:321 (qt_internal_get_darwin_sdk_version)
  cmake/QtAutoDetect.cmake:483 (qt_auto_detect_darwin)
  CMakeLists.txt:18 (include)
```

Reply:
Done. Untested on x86_64-darwin though.